examples/tutorials/vectors.py

```diff
@@ -0,0 +1,486 @@
+# functions are used. The first of which is ``np.arange`` which iterates from 0
+# to the value of ct. This function is used to generate random data points for
+# ``radius`` and ``stopdir`` since these values are intended to be different in
+# each of the five vectors begin plotted. The second function is ``np.full``
+# which creates a ``numpy.ndarray``. Finally, all of this data is congregated into
+# a final ``numpy.ndarray`` called data using ``np.column_stack``. This is then
+# passed to the plot function and the resulting figure is shown below.
+
+ct = 5
+radius = 3 - (0.5 * np.arange(0, ct))
+startdir = np.full(ct, 90)
+stopdir = 180 + (50 * np.arange(0, ct))
+
+data = np.column_stack([np.full(ct, 0), np.full(ct, 0), radius, startdir, stopdir])
+
+fig = pygmt.Figure()
+fig.plot(
+    region=[-5, 5, -5, 5],
+    projection="X10c",
+    frame="ag",
+    data=data,
+    style="m0.5c+ea",
+    pen="2p",
+    color="red3",
+)
+fig.show()
+
+########################################################################################
+# Much like when plotting regular vectors, the default unit used is centimeters.
+# When this is changed to inches, the size of the plot appears larger when the
+# projection units do not change. Below is an example of two circular vectors.
+# One is plotted using the default unit, and the second is plotted using inches.
+# The difference in size of the two vectors provides good insight into how this
+# functionality works.
+
+circular_vector_1 = [6, 5, 2, 90, 270]
+circular_vector_2 = [6, 5, 1, 90, 270]
+
+data_1 = [circular_vector_1]
+
+fig = pygmt.Figure()
+fig.plot(
+    region=[0, 10, 0, 10],
+    projection="X10c",
+    frame="ag",
+    data=data_1,
+    style="m0.5c+ea",
+    pen="2p",
+    color="red3",
+)
+
+data_2 = [circular_vector_2]
+with pygmt.config(PROJ_LENGTH_UNIT="i"):
+    fig.plot(
+        data=data_2,
+        style="m0.5c+ea",
+        pen="2p",
+        color="red3",
+    )
+fig.show()
+
+########################################################################################
+# Plot Geographic Vectors
+# -----------------------
+# Geographic graph using x and y values to set a start and an ending point.
+# Use ``fig.coast`` to plot a coast. ``x`` and ``y`` are coordinates
+# on a grid that we are using. ``x`` is Idaho and ``y`` is Chicago in this example.
+# The geographical vector is going from Idaho to Chicago. To style geographic
+# vectors, use ``=`` at the begining of the ``style`` parameter.
+# Other styling features such as arrow head color and line thickness
+# can be passed into
+# ``pen`` and ``color``parameters
+
+fig = pygmt.Figure()
+fig.coast(
+    region=[-127, -64, 24, 53],
+    projection="M10c",
+    frame=True,
+    borders=1,
+    shorelines="0.25p,black",
+    area_thresh=4000,
+)
+point_1 = [-114.7420, 44.0682]
+point_2 = [-87.6298, 41.8781]
+data = np.array([point_1 + point_2])
+
+fig.plot(
+    data=data,
+    style="=0.5c+ea+s",
```

Comment:
`+s` is very important. Please explain it in the introduction text below the title "Plot Geographic Vectors". See [Vector Attributes](https://docs.generic-mapping-tools.org/latest/plot.html#vector-attributes) in the GMT documentation.
> In addition, all but circular vectors may take these modifiers:
> - +s means the input angle, length are instead the xe,ye coordinates of the vector end point.

---

For cartesian vectors, you don't show an example to use the `+s` argument. How do you think adding one?

---

It seems that you only show one way to indicate the geographic vectors. But we can still use **Azimuth** (in degrees east from north) and geographical length, right? Shall we add an example? See [-S=](https://docs.generic-mapping-tools.org/latest/plot.html#s) in GMT documentation.
> Geographic vector. Azimuth (in degrees east from north) and geographical length must be found in columns 3 and 4. The size is the length of the vector head. Vector width is set by -W. See Vector Attributes for specifying attributes. Note: Geovector stems are drawn as thin filled polygons and hence pen attributes like dashed and dotted are not available. For allowable geographical units, see Units.

Reply:
> `+s` is very important. Please explain it in the introduction text below the title "Plot Geographic Vectors". See [Vector Attributes](https://docs.generic-mapping-tools.org/latest/plot.html#vector-attributes) in the GMT documentation.
> 
> > In addition, all but circular vectors may take these modifiers:
> > 
> > * +s means the input angle, length are instead the xe,ye coordinates of the vector end point.

Yes, one-sentence explanation should be good.

> For cartesian vectors, you don't show an example to use the `+s` argument. How do you think adding one?
> 
> It seems that you only show one way to indicate the geographic vectors. But we can still use **Azimuth** (in degrees east from north) and geographical length, right? Shall we add an example? See [-S=](https://docs.generic-mapping-tools.org/latest/plot.html#s) in GMT documentation.
> 
> > Geographic vector. Azimuth (in degrees east from north) and geographical length must be found in columns 3 and 4. The size is the length of the vector head. Vector width is set by -W. See Vector Attributes for specifying attributes. Note: Geovector stems are drawn as thin filled polygons and hence pen attributes like dashed and dotted are not available. For allowable geographical units, see Units.

I prefer not to make the tutorial too long to follow.

Reply:
I'm about to add the sentence clarifying `+s`, however, I agree with @seisman that this tutorial and PR is getting too long. If it's needed, I'd be happy to add another example for cartesian and geographic vectors using `+s` and  `-S=` later in a different PR. I hope that this is ok with you.

Reply:
Okay~ :smile: